### PR TITLE
v0.40 Deprecations

### PR DESCRIPTION
**Pull request by banesullivan**
Deprecations marked for removal on v0.40

I recommend waiting to merge these until the dust settles as this PR could muck up any dire patch releases.

### CODE REVIEW

File: pyvista/core/grid.py [hunk 1]
```diff
@@ -485,70 +484,29 @@ class UniformGrid(_vtk.vtkImageData, Grid, UniformGridFilters):
 
     def __init__(
         self,
-        uinput=None,
+        image=None,
```
Comment (banesullivan) [inline pyvista/core/grid.py:487]: Maybe we shouldn't change the name of this argument?

Comment (adeak) [reply pyvista/core/grid.py:487]: On a related note, in half a year when we drop 3.7 support we should consider making all these parameters [positional-only](https://peps.python.org/pep-0570/):
```py
>>> def foo(self, bar=None, /, *args, baz=None):
...     print(bar)
... 
>>> foo(3)
None
>>> foo(3, bar=42)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: foo() got some positional-only arguments passed as keyword arguments: 'bar'
```
This has the same risk as renaming it today, hence related note.

Comment (adeak) [reply pyvista/core/grid.py:487]: I guess we could do the same breaking today by removing the argument and doing `image, *args = args` in the method body, but then we'd have to undo that when we can use positional-only parameters for more readability.

Comment (banesullivan) [reply pyvista/core/grid.py:487]: We already dropped 3.7 (ref https://github.com/pyvista/pyvista/pull/3938) and 

https://github.com/pyvista/pyvista/blob/9ee35ba478e573711f54b3fdac5fb04242208c22/pyproject.toml#L12

Comment (adeak) [reply pyvista/core/grid.py:487]: Oh, nice, I forgot about that. In that case definitely consider making it positional-only if we rename it.

**Comment by banesullivan** [comment]
Too many conflicts. Recommend new PR

**Pull request closed without merging**